Clarify documentation regarding Climate entity hvac_action (#3270)
home-assistant/developers.home-assistant: docs/core/entity/climate.md

### HVAC modes

The HVAC mode is the behaviour that the device is requested to perform.

You are only allowed to use the built-in HVAC modes, provided by the `HVACMode`

> enum. For behaviour that is not covered by the HVAC modes, add a preset instead.

Reviewer: This sounds like presets could be a replacement for a mode which is not the case

Author: Isn't the original text "If you want another mode, add a preset instead." worse, then?

Reviewer: I didn't say anything else, Maybe something like `For behavior that is not covered by the HVAC modes, it can be extended by using presets` (as you still have to have a mode even if you're using presets)

### HVAC action

> The HVAC action describes the _current_ action that the device is performing in order to fulfill the requested HVAC mode as determined by the device's thermostat, humidistat, and control algorithms.

Reviewer: These examples sounds more like rules rather than examples of use. They are also a lot of repetition of what's already in the table.
Maybe we can just keep the table and extend the description a bit where it makes sense to add another layer of explanation (and/or example)?

Author: I was a bit on the fence of whether to include the examples, but in the end I decided to include them in the PR since it seemed like it would be easier to understand than trying to explain what the actions actually mean.

The goal was to avoid a misinterpretation that I had witnessed where another developer thought that an air conditioner should be "idle" once the target temperature is reached, even if the air conditioner is still performing work to remove heat from the room with the goal of maintaining the room at that temperature.

My original idea brought in some physics to explain why "heating" and "cooling" are not the same thing as "causing temperature to increase" and "causing temperature to decrease". When I got to the point where I was thinking about mentioning the difference between of sensible heat vs. latent heat of vaporization to explain why dehumidifiers cause the temperature to increase, I figured it was probably hurting more than helping. Go watch a [Technology Connections video](https://www.youtube.com/watch?v=j_QfX0SYCE8) instead :)

So instead of describing the physics explanation _why_ of when certain actions are applicable, I wrote some examples which I think cover the same ground. The goal was to clarify when it was appropriate to use "idle", and demonstrate situations where a device can still be "heating" or "cooling" even after the target temperature has been reached.

Perhaps to make the docs easier to scan, I could move the examples after the table?

Reviewer: We can try that to have the table first, and limit the examples to those that is "needed", as example, `defrosting` or `off` is hardly needed to explain but perhaps `heating`, `cooling` and `idle` as you already mentioned could have a sentence.

### HVAC action

The HVAC action describes the _current_ action that the device is performing in order to fulfill the requested HVAC mode and preset,
as determined by the device's own control algorithms.

You are only allowed to use the built-in HVAC actions provided by the `HVACAction` enum.

:::note
It might not be possible to accurately determine the correct HVAC action unless the device reports additional information.
For example:

- A thermostat with hysteresis has an ambiguous temperature range near the target temperature where it may be performing an action or may be idle.
- A variable power device might continue to operate at a reduced power level after the target temperature is reached, rather than becoming idle.

> For devices where the HVAC action cannot be accurately determined, use the default `None`.

Reviewer: ```suggestion
For devices where the HVAC action cannot be accurately determined, do not implement `hvac_action`.
```
